tests/sentry/api/endpoints/test_user_authenticator_enroll.py

```diff
@@ -254,6 +256,49 @@ def test_u2f_disallow_new_enrollment(self):
             },
         )
 
+    @mock.patch("sentry.auth.authenticators.U2fInterface.try_enroll", return_value=True)
+    @override_options({"system.url-prefix": "https://testserver"})
+    def test_u2f_superuser_and_staff_cannot_enroll_other_user(self, try_enroll):
+        elevated_user = self.create_user(is_superuser=True, is_staff=True)
+        self.login_as(user=elevated_user, superuser=True, staff=True)
+
+        resp = self.get_success_response(self.user.id, "u2f")
```

Comment:
hmm shouldn't this error? if a superuser can't enroll u2f for another user, why should hitting the endpoint succeed and instead enroll that u2f for the superuser?

Reply:
or am i misunderstanding

Reply:
The existing code is like that, I just added a test for it. I do agree that the continuing with the logic seems a bit odd

Reply:
Relevant comment:
https://github.com/getsentry/sentry/blob/c6cc55c7f13f37ce8b9e3646a955f2cacac650e8/src/sentry/api/endpoints/user_authenticator_enroll.py#L190-L192

Reply:
can we add a TODO to investigate this behavior? it seems funky, but i don't think it's in our scope to go into it right now